examples/acados_python/pendulum_example/minimal_example_closed_loop.py

```diff
@@ -0,0 +1,142 @@
+# Andrea Zanelli, Niels van Duijkeren, Jonathan Frey, Tommaso Sartor,
+# Branimir Novoselnik, Rien Quirynen, Rezart Qelibari, Dang Doan,
+# Jonas Koenemann, Yutao Chen, Tobias Schöls, Jonas Schlagenhauf, Moritz Diehl
+#
+# This file is part of acados.
+#
+# The 2-Clause BSD License
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.;
+#
+
+from acados_template import *
+from export_pendulum_ode_model import export_pendulum_ode_model
+from utils import plot_pendulum
+import numpy as np
+import scipy.linalg
+
+FORMULATION = 'LS' # 'LS'
+
+# create ocp object to formulate the OCP
+ocp = acados_ocp_nlp()
+
+# set model
+model = export_pendulum_ode_model()
+ocp.model = model
+
+Tf = 1.0
+nx = model.x.size()[0]
+nu = model.u.size()[0]
+ny = nx + nu
+ny_e = nx
+N = 20
+
+# set dimensions
+ocp.dims.nx  = nx
+ocp.dims.ny  = ny
+ocp.dims.ny_e = ny_e
+ocp.dims.nbx = 0
+ocp.dims.nbu = nu
+ocp.dims.nu  = nu
+ocp.dims.N   = N
+
+# set cost module
+ocp.cost.cost_type = 'LINEAR_LS'
+ocp.cost.cost_type_e = 'LINEAR_LS'
+
+Q = 2*np.diag([1e3, 1e3, 1e-2, 1e-2])
+R = 2*np.diag([1e-2])
+
+ocp.cost.W = scipy.linalg.block_diag(Q, R)
+
+ocp.cost.W_e = Q
+
+ocp.cost.Vx = np.zeros((ny, nx))
+ocp.cost.Vx[:nx,:nx] = np.eye(nx)
+
+Vu = np.zeros((ny, nu))
+Vu[4,0] = 1.0
+ocp.cost.Vu = Vu
+
+ocp.cost.Vx_e = np.eye(nx)
+
+ocp.cost.yref  = np.zeros((ny, ))
+ocp.cost.yref_e = np.zeros((ny_e, ))
+
+# set constraints
+Fmax = 80
+x0 = np.array([0.0, np.pi, 0.0, 0.0])
+ocp.constraints.constr_type = 'BGH'
+ocp.constraints.lbu = np.array([-Fmax])
+ocp.constraints.ubu = np.array([+Fmax])
+ocp.constraints.x0 = x0
+ocp.constraints.idxbu = np.array([0])
+
+ocp.solver_options.qp_solver = 'PARTIAL_CONDENSING_HPIPM'
```

Comment:
There is a trade off.
On the one hand this examples should be short and understandable.
On the other hand they should show case well enough how the interface works and show its flexibility.

Reply:
For the zero dimensions, I definitely agree and removed these assignments.
Feel free to minimize the examples further.

Reply:
It would would be cool to say that we have at least one minimal example with very few lines of code. Somehow people tend to be affectionate to this parameter when comparing `acados` and `ACADO` for example  :P.

Reply:
let's do this anyway later on.